src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/rca/hotshard/HighCPUShardRca.java

```diff
@@ -0,0 +1,209 @@
+
+    //TODO : {@khushbr} refine the threshold values and read same from config file
+    private static final double CPU_USAGE_THRESHOLD = 0.01;
+    private static final double IO_TOT_THROUGHPUT_THRESHOLD_IN_BYTES = 250000;
+    private static final double IO_TOT_SYSCALL_RATE_THRESHOLD_PER_SECOND = 0.01;
+
+    private final Metric cpuUsage;
+    private final Metric ioTotThroughput;
+    private final Metric ioTotSyscallRate;
+    private final ResourceType resourceType;
+    private final int rcaPeriod;
+    private int counter;
+    protected Clock clock;
+
+    // HashMap with IndexShardKey object as key and SlidingWindowData object of metric data as value
+    private HashMap<IndexShardKey, SlidingWindow<SlidingWindowData>> cpuUsageHashMap;
+    private HashMap<IndexShardKey, SlidingWindow<SlidingWindowData>> ioTotThroughputHashMap;
+    private HashMap<IndexShardKey, SlidingWindow<SlidingWindowData>>  ioTotSyscallRateHashMap;
+
+    public <M extends Metric> HighCPUShardRca(final long evaluationIntervalSeconds, final int rcaPeriod,
+            final M cpuUsage, final M ioTotThroughput, final M ioTotSyscallRate) {
+        super(evaluationIntervalSeconds);
+        this.cpuUsage = cpuUsage;
+        this.ioTotThroughput = ioTotThroughput;
+        this.ioTotSyscallRate = ioTotSyscallRate;
+        this.rcaPeriod = rcaPeriod;
+        this.counter = 0;
+        this.resourceType = ResourceType.newBuilder().setHardwareResourceTypeValue(HardwareEnum.CPU_VALUE).build();
+        this.clock = Clock.systemUTC();
+        this.cpuUsageHashMap = new HashMap<>();
+        this.ioTotThroughputHashMap = new HashMap<>();
+        this.ioTotSyscallRateHashMap = new HashMap<>();
+    }
+
+    private void consumeFlowUnit(MetricFlowUnit metricFlowUnit, String metricType,
+                                 HashMap<IndexShardKey, SlidingWindow<SlidingWindowData>> metricMap) {
+        for (Record record : metricFlowUnit.getData()) {
+            try {
+                String indexName = record.getValue(INDEX_NAME.toString(), String.class);
+                String shardId = record.getValue(SHARD_ID.toString(), String.class);
+                if (indexName == null || shardId == null) {
+                    continue;
+                }
+                double usage = record.getValue(MetricsDB.SUM, Double.class);
+                IndexShardKey indexShardKey = new IndexShardKey(indexName, shardId);
+                SlidingWindow<SlidingWindowData> usageDeque = metricMap.get(indexShardKey);
+                if (null == usageDeque) {
+                    usageDeque = new SlidingWindow<>(SLIDING_WINDOW_IN_SECONDS, TimeUnit.SECONDS);
+                    metricMap.put(indexShardKey, usageDeque);
+                }
+                usageDeque.next(new SlidingWindowData(this.clock.millis(), usage));
+            } catch (Exception e) {
+                // TODO: Add a metric here.
+                LOG.error("Failed to parse metric in FlowUnit from {}", metricType);
```

Comment:
Should we also add the record here so that looking at the logs we will infer why ?

Reply:
Done.